queryscript/src/compile/generics.rs

```diff
@@ -201,8 +205,28 @@ impl Generic for SumGeneric {
         })?;
         Ok(())
     }
+
+    fn resolve(&self) -> Result<CRef<MType>> {
+        let this = self.clone();
+        this.0.clone().then(move |_: Ref<MType>| {
+            let rt = match this.to_runtime_type().context(RuntimeSnafu {
+                loc: ErrorLocation::Unknown,
+            }) {
+                Ok(rt) => rt,
+                Err(_) => {
+                    return Ok(mkcref(MType::Generic(Located::new(
+                        Arc::new(this.clone()),
+                        SourceLocation::Unknown,
+                    ))))
```

Comment:
I don't follow this logic — if it's not ready to be a runtime type yet, we return ourselves? Is this infinitely recursive?

Reply:
Also, it's not infinitely recursive because `resolve_generics` isn't doing anything to re-run over its own output or anything like that.

Reply:
Okay I understand now (and appreciate the comment)